plugins/main/public/components/common/wazuh_discover/wz_discover.tsx

```diff
@@ -0,0 +1,247 @@
+  const SearchBar = getPlugins().data.ui.SearchBar;
+  const [results, setResults] = useState<SearchResponse>({} as SearchResponse);
+  const [inspectedHit, setInspectedHit] = useState<any>(undefined);
+  const [indexPattern, setIndexPattern] = useState<IndexPattern | undefined>(undefined);
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+  const [isExporting, setIsExporting] = useState<boolean>(false);
+
+  const onClickInspectDoc = useMemo(() => (index: number) => {
+    const rowClicked = results.hits.hits[index];
+    setInspectedHit(rowClicked);
+  }, [results]);
+
+  const DocViewInspectButton = ({ rowIndex }: EuiDataGridCellValueElementProps) => {
+    const inspectHintMsg = 'Inspect document details';
+    return (
+      <EuiToolTip content={inspectHintMsg}>
+        <EuiButtonIcon
+          onClick={() => onClickInspectDoc(rowIndex)}
+          iconType='inspect'
+          aria-label={inspectHintMsg}
+        />
+      </EuiToolTip>
+    );
+  };
+
+  const dataGridProps = useDataGrid({
+    ariaLabelledBy: 'Vulnerabilities Inventory Table',
+    defaultColumns: defaultTableColumns,
+    results,
+    indexPattern: indexPattern as IndexPattern,
+    DocViewInspectButton
+  })
+
+  const { pagination, sorting, columnVisibility } = dataGridProps;
+
+  const docViewerProps = useDocViewer({
+    doc: inspectedHit,
+    indexPattern: indexPattern as IndexPattern,
+  })
+
+  useEffect(() => {
+    if (!isLoading) {
+      setIndexPattern(indexPatterns?.[0] as IndexPattern);
+      search({
+        indexPattern: indexPatterns?.[0] as IndexPattern,
+        filters,
+        query,
+        pagination,
+        sorting
+      }).then((results) => {
+        setResults(results);
+        setIsSearching(false);
+      }).catch((error) => {
+        const searchError = ErrorFactory.create(HttpError, { error, message: 'Error fetching vulnerabilities' })
+        ErrorHandler.handleError(searchError);
+        setIsSearching(false);
+      })
+    }
+  }, [JSON.stringify(searchBarProps), JSON.stringify(pagination), JSON.stringify(sorting)]);
+
+  const timeField = indexPattern?.timeFieldName ? indexPattern.timeFieldName : undefined;
+
+  const onClickExportResults = async () => {
+    const params = {
+      indexPattern: indexPatterns?.[0] as IndexPattern,
+      filters,
+      query,
+      fields: columnVisibility.visibleColumns,
+      pagination: {
+        pageIndex: 0,
+        pageSize: results.hits.total
+      },
+      sorting
+    }
+    try {
+      setIsExporting(true);
+      await exportSearchToCSV(params);
+    } catch (error) {
+      const searchError = ErrorFactory.create(HttpError, { error, message: 'Error downloading csv report' })
+      ErrorHandler.handleError(searchError);
+    } finally {
+      setIsExporting(false);
+    }
+  }
+
+  return (
+    <IntlProvider locale="en">
+      <EuiPageTemplate
+        className="vulsInventoryContainer"
```

Comment:
Change class name as it will be used in other modules besides the Vulnerability module.

Reply:
Thanks @chantal-kelm